internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -1555,44 +1711,133 @@ func resourceOrchestratedVirtualMachineScaleSetDelete(d *pluginsdk.ResourceData,
 	return nil
 }
 
-func expandOrchestratedVirtualMachineScaleSetSkuProfile(input []interface{}) *virtualmachinescalesets.SkuProfile {
+func expandOrchestratedVirtualMachineScaleSetSkuProfile(d *pluginsdk.ResourceData, input []interface{}) *virtualmachinescalesets.SkuProfile {
 	if len(input) == 0 || input[0] == nil {
 		return nil
 	}
 
 	v := input[0].(map[string]interface{})
-	vmSizesRaw := v["vm_sizes"].(*pluginsdk.Set).List()
+	allocationStrategy := v["allocation_strategy"].(string)
 	vmSizes := make([]virtualmachinescalesets.SkuProfileVMSize, 0)
-	for _, vmSize := range vmSizesRaw {
-		vmSizes = append(vmSizes, virtualmachinescalesets.SkuProfileVMSize{
-			Name: pointer.To(vmSize.(string)),
-		})
+
+	if !features.FivePointOh() {
+		vmSizesRaw := v["vm_sizes"].(*pluginsdk.Set).List()
+		vmSizeRaw := v["vm_size"].(*pluginsdk.Set).List()
+
+		if len(vmSizesRaw) > 0 {
+			for _, vmSize := range vmSizesRaw {
+				vmSizes = append(vmSizes, virtualmachinescalesets.SkuProfileVMSize{
+					Name: pointer.To(vmSize.(string)),
+				})
+			}
+		} else if len(vmSizeRaw) > 0 {
+			for i, vmSizeRaw := range vmSizeRaw {
+				vmSizeMap := vmSizeRaw.(map[string]interface{})
+				vmSize := virtualmachinescalesets.SkuProfileVMSize{
+					Name: pointer.To(vmSizeMap["name"].(string)),
+				}
+
+				if rawConfig := d.GetRawConfig(); !rawConfig.IsNull() {
+					if skuProfile := rawConfig.AsValueMap()["sku_profile"]; !skuProfile.IsNull() {
+						if skuProfiles := skuProfile.AsValueSlice(); len(skuProfiles) > 0 && !skuProfiles[0].IsNull() {
+							if vmSizePath := skuProfiles[0].AsValueMap()["vm_size"]; !vmSizePath.IsNull() {
+								if vmSizes := vmSizePath.AsValueSlice(); len(vmSizes) > i && !vmSizes[i].IsNull() {
+									if rank := vmSizes[i].AsValueMap()["rank"]; !rank.IsNull() {
+										vmSize.Rank = pointer.To(int64(vmSizeMap["rank"].(int)))
+									}
+								}
+							}
+						}
+					}
+				}
+
+				vmSizes = append(vmSizes, vmSize)
+			}
+		}
+	} else {
```

Comment:
could we simplify this by removing the `else if` section from this? we could just have this as one block of code below if `vm_size` > 0

Reply:
Yes, I think that would work since the schema makes sure that `vm_size` and `vm_sizes` are mutually exclusive, so you can only define one or the other but not both. So if `vm_size` or `vm_sizes` was not in the config, it would return an list of 0 length, which means the for loop would be a no-op, looping from 0 to 0 and it would basically sikp the code in the loop entirely?